scripts/newtab.js

```diff
@@ -0,0 +1,188 @@
+'use strict'
+
+var placeholders = [];
+var url = 'https://bwinton.github.io/whimsy/thumbnail-gifs.txt';
+var getting = browser.storage.sync.get('newtab');
+var listener = false;
+var tippyTopSites = require('tippy-top-sites');
+
+initialize();
+
+function onError(error) {
+  console.log(error);
+}
+function loadTopSites(){
+  //display top sites
+  browser.topSites.get().then((topSitesArray) => {
+    var grid = document.querySelector("#newtab-grid");
+    grid.innerHTML = "";
+    for (let topSite of topSitesArray) {
+      grid.innerHTML +=
+      "<div class='newtab-cell'><div class='newtab-site' draggable='true' type='history'>"+
+          "<a class='newtab-link' title="+topSite.title+" href="+topSite.url+">"+
+            "<span class='newtab-thumbnail thumbnail' style='background-size: cover; background-image: ;'></span>"+
+            "<span class='newtab-title'>"+topSite.title+"</span>"+
+          "</a>"+
+        "</div>"+
+      "</div>"
+    }
+  }, onError);
+}
+function initialize(){
+  fetch(url).then(function(response){
```

Comment:
So, this is the second place we have code like this, if I remember correctly.  Can we extract it into a utility function in another file?